### PR TITLE
Unit deploy to

### PR DESCRIPTION
**Pull request by huwshimi**
Allow placing of unplaced units via a form.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- use `/:flags:/mv/`
- add a service
- open the machine view
- on the unplaced machine click "Deploy to..."
- use the forms in various combinations to place units

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3934/
Test PASSed.

File: jujugui/static/gui/src/app/components/machine-view/add-machine/add-machine.js [hunk 1]
```diff
@@ -53,18 +57,44 @@ YUI.add('machine-view-add-machine', function() {
...
+      var constraints = state.constraints;
+      var machine;
+      if (machineId && machineId !== 'new' && !selectedContainer) {
+        // Don't try and create a container if the container type has not been
+        // selected.
         return;
       }
-      this.props.createMachine(
-        this.state.containerType, this.props.parentId, this.state.constraints);
+      if (selectedMachine === 'new' || selectedContainer === 'lxc' ||
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/machine-view/add-machine/add-machine.js]: We probably won't remember why these conditionals are here (and the `props.unit` one below) Could you add comments to the rational here. Thankx

**Comment by hatched** [comment]
:+1: :ok: Just one request for some comments! Thanks!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3940/
Test PASSed.

**Comment by huwshimi** [comment]
Thanks @hatched :shipit:!

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**